File: common/src/main/java/com/ultreon/mods/advanceddebug/client/menu/pages/WindowPage.java
package com.ultreon.mods.advanceddebug.client.menu.pages;

import com.mojang.blaze3d.platform.Window;
import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage;
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext;
import com.ultreon.mods.advanceddebug.mixin.common.OptionsAccessor;
import com.ultreon.mods.advanceddebug.util.InputUtils;
import net.minecraft.client.gui.GuiGraphics;
import org.jetbrains.annotations.NotNull;

public class WindowPage extends DebugPage {
    public WindowPage() {

    }

    @Override
    public void render(@NotNull GuiGraphics gfx, IDebugRenderContext ctx) {
        Window window = getWindow();

        ctx.left("Scale / Size");
        ctx.left("Gui Scale", getMultiplier(window.getGuiScale()));
        ctx.left("Window Size", getSize(window.getScreenWidth(), window.getScreenHeight()));
        ctx.left("Window Size (Scaled)", getSize(window.getGuiScaledWidth(), window.getGuiScaledHeight()));
        ctx.left("Framebuffer Size", getSize(window.getWidth(), window.getHeight()));
        ctx.left();

        ctx.right("Misc");
        ctx.right("Refresh Rate", window.getRefreshRate());
        ctx.right("Framerate Limit", window.getFramerateLimit());
        ctx.right("Fullscreen Mode", window.isFullscreen());
        ctx.right("Vsync", ((OptionsAccessor) minecraft.options).getEnableVsync().get());

        if (InputUtils.isAltDown()) {
            ctx.top("Please don't press Alt+F4"); // Just don't.
        }
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/client/menu/pages/WindowPage.kt
package com.ultreon.mods.advanceddebug.client.menu.pages

import com.mojang.blaze3d.vertex.PoseStack
import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext
import com.ultreon.mods.advanceddebug.api.common.multiplier
import com.ultreon.mods.advanceddebug.api.common.size
import com.ultreon.mods.advanceddebug.mixin.common.OptionsAccessor
import com.ultreon.mods.advanceddebug.util.InputUtils

class WindowPage(modId: String, name: String) : DebugPage(modId, name) {
    override fun render(poseStack: PoseStack, ctx: IDebugRenderContext) {
        ctx.left("Scale / Size")
        ctx.left("Gui Scale", mainWindow.guiScale.multiplier)
        ctx.left("Window Size", size(mainWindow.screenWidth, mainWindow.screenHeight))
        ctx.left("Window Size (Scaled)", size(mainWindow.guiScaledWidth, mainWindow.guiScaledHeight))
        ctx.left("Framebuffer Size", size(mainWindow.width, mainWindow.height))
        ctx.left()
        ctx.right("Misc")
        ctx.right("Refresh Rate", mainWindow.refreshRate)
        ctx.right("Framerate Limit", mainWindow.framerateLimit)
        ctx.right("Fullscreen Mode", mainWindow.isFullscreen)
        ctx.right("Vsync", (minecraft.options as OptionsAccessor).enableVsync)
        if (InputUtils.isAltDown) {
            ctx.top("Please don't press Alt+F4") // Just don't.
        }
    }
}